workflow/installertest.groovy

```diff
@@ -0,0 +1,133 @@
+// Replace colons in image with hyphens and append text
+String convertImageNameToString(String imageName, String append="") {
+    return (imageName+append).replaceAll(':','-')
+}
+
+/**
+ * Extracts the components from an {@code 'artifact://full/path/to/job/buildNr#artifact.ext'} type url.
+ * @param url the url
+ */
+@NonCPS
+def getComponentsFromArtifactUrl(String url) {
```

Comment:
Or just fix the bugs in groovy sandbox and cps so you can use standard groovy regexp :p